### PR TITLE
Add `frem`, `fptoui`, `uitofp` instructions to QIR simulators

### PR DESCRIPTION
**Pull request by orpuente-MS**
This PR adds support for the `frem`, `fptoui`, and `uitofp` instructions to the QIR simulators to handle the codegen changes in #3187 .

### CODE REVIEW

File: source/qdk_package/tests/test_adaptive_cpu_bytecode.py [hunk 1]
```diff
@@ -784,6 +784,11 @@ def test_fcmp(sim_type, pred, lhs, rhs, expected):
     ("fsub", 3.0, 10.0, -7.0),
     ("fmul", 6.0, 7.0, 42.0),
     ("fdiv", 8.0, 2.0, 4.0),
+    ("frem", 8.0, 2.0, 0.0),
```
Comment (joao-boechat) [inline source/qdk_package/tests/test_adaptive_cpu_bytecode.py:787]: Should we have test cases where divisor is 0 to make sure it fails gracefully?

Comment (joao-boechat) [reply source/qdk_package/tests/test_adaptive_cpu_bytecode.py:787]: Also would be nice to test fptoui with negative floats

File: source/simulators/src/gpu_full_state_simulator/simulator_adaptive.wgsl [hunk 2]
```diff
@@ -1210,6 +1213,16 @@ fn interpret_classical(@builtin(global_invocation_id) gid: vec3<u32>) {
                 pc++;
             }
 
+            // FREM: Float remainder. LLVM docs say this instruction has
+            // the same semantics as C's fmod, which is implemented as:
+            // dst = src0 - trunc(src0/src1) * src1
+            case OP_FREM {
+                let a = resolve_f32(shot_idx, instr.src0, flags, 0u);
+                let b = resolve_f32(shot_idx, instr.src1, flags, 1u);
```
Comment (joao-boechat) [inline source/simulators/src/gpu_full_state_simulator/simulator_adaptive.wgsl:1221]: one thing that copilot noted and I thought was interesting: the GPU uses f32, while CPU uses f64. Should we standardize that or write this down somewhere so we don't introduce bugs in the future?

**Review by joao-boechat — APPROVED** [review]

File: source/simulators/src/gpu_full_state_simulator/simulator_adaptive.wgsl (hunk 2 above)
Comment (orpuente-MS) [reply source/simulators/src/gpu_full_state_simulator/simulator_adaptive.wgsl:1221]: This is a known limitation of the GPU simulator, and the QIR -> bytecode compiler follows different logic depending on if we are compiling for a 32-bit target (the CPU simulators) or a 64-bit target (the GPU simulators).

File: source/qdk_package/tests/test_adaptive_cpu_bytecode.py (hunk 1 above)
Comment (orpuente-MS) [reply source/qdk_package/tests/test_adaptive_cpu_bytecode.py:787]: Yes, nice catch. I added those tests to the CPU and GPU simulators.
